### PR TITLE
Adds SYNC-3148 [v108] Include history apis in RustPlaces

### PR DESCRIPTION
**Pull request by tarikeshaq**
Adds the history APIs to RustPlaces.swift

This is a small PR to add the history APIs to RustPlaces. It doesn't change or introduce any new behaviour.

This will be followed by https://github.com/tarikeshaq/firefox-ios/pull/6 which has the bulk of the behaviour changes for the History

cc @nbhasin2 @lougeniaC64 @mhammond for review

### CODE REVIEW

**Review by lougeniaC64 — APPROVED** [review]
LGTM 🚀

File: Storage/Rust/RustPlaces.swift [hunk 1]
```diff
@@ -463,3 +509,76 @@ public class RustPlaces: BookmarksHandler {
         }
     }
 }
+
+// MARK: History APIs
+
+extension VisitTransition {
+    static public func fromVisitType(visitType: VisitType) -> Self {
```
Comment (nbhasin2) [inline Storage/Rust/RustPlaces.swift:516]: nit: 
```suggestion
    public static func fromVisitType(visitType: VisitType) -> Self {
```

**Review by nbhasin2 — APPROVED** [review]

**Comment by tarikeshaq** [comment]
@nbhasin2 I don't think the test failures are related (even though they say "HistoryTests" 🙃) This PR only introduces functions and doesn't adjust any behaviour

**Comment by tarikeshaq** [comment]
And otherwise this should be ready to land (I don't have permission :()

**Comment by lmarceau** [comment]
Build is green [here](https://app.bitrise.io/app/6c06d3a40422d10f/pipelines/39d9c4c9-6151-49e8-be97-245eeac63c3d)
Screen Shot 2022-10-20 at 4 46 24 PM

**Comment by nbhasin2** [comment]
@tarikeshaq landed, thanks for this :)

**Pull request merged**